src/core/src/op/loop.cpp

```diff
@@ -225,11 +232,38 @@ void op::v5::Loop::validate_and_infer_types() {
             if (ps.is_dynamic()) {
                 set_output_type(index, body_value.get_element_type(), ps);
             } else {
-                auto shape = ps.get_shape();
-                if (zero_number_of_iter) {
-                    shape.at(0) = 0;
+                if (back_edges.count(output_description->m_body_value_index)) {
```

Comment:
I think new piece of code below should be applicable for the dynamic shapes as well as for static ones.

Also please put more comments in the code to understand the purpose of this part.

Reply:
Add the dynamic case and more comments.